examples/todo-intermediate/test/acceptance/todo-list-todo.acceptance.ts

```diff
@@ -0,0 +1,164 @@
+import {createClientForHandler, expect, supertest} from '@loopback/testlab';
+import {TodoListApplication} from '../../src/application';
+import {Todo, TodoList} from '../../src/models/';
+import {TodoRepository, TodoListRepository} from '../../src/repositories/';
+import {givenTodo, givenTodoList} from '../helpers';
+
+describe('Application', () => {
+  let app: TodoListApplication;
+  let client: supertest.SuperTest<supertest.Test>;
+  let todoRepo: TodoRepository;
+  let todoListRepo: TodoListRepository;
+
+  before(givenRunningApplicationWithCustomConfiguration);
+  after(() => app.stop());
+
+  before(givenTodoRepository);
+  before(givenTodoListRepository);
+  before(() => {
+    client = createClientForHandler(app.requestHandler);
+  });
+
+  // TODO: refactor into multiple files
+  beforeEach(async () => {
+    await todoRepo.deleteAll();
+    await todoListRepo.deleteAll();
+  });
+
+  it('creates todo for a todoList', async () => {
+    const todoList = await givenTodoListInstance();
+    const todo = givenTodo();
+    const response = await client
+      .post(`/todo-lists/${todoList.id}/todos`)
+      .send(todo)
+      .expect(200);
+
+    expect(response.body).to.containDeep(todo);
+  });
+
+  it('finds todos for a todoList', async () => {
+    const todoList = await givenTodoListInstance();
+    const notMyTodo = await givenTodoInstance({
+      title: 'someone else does a thing',
+    });
+    const myTodos = [
+      await givenTodoInstanceOfTodoList(todoList.id),
+      await givenTodoInstanceOfTodoList(todoList.id, {
+        title: 'another thing needs doing',
+      }),
+    ];
+    const response = await client
+      .get(`/todo-lists/${todoList.id}/todos`)
+      .send()
+      .expect(200);
+
+    expect(response.body)
+      .to.containDeep(myTodos)
+      .and.not.containEql(notMyTodo.toJSON()); // is this assertion necessary?
```

Comment:
@bajtos Do you feel it is up to the users to test whether the queries have been properly constrained here? This line for example, I assert that the models discovered by the constrained repository's `find` does not contain a todo with a different foreign key that the repository's been constrained with. Is this assertion unnecessary since it's something that the users should trust with the framework?

Reply:
In acceptance tests, I'd prefer to verify that the application works correctly end-to-end, i.e. verify assumptions made about the framework too.

On the other hand, if we had an integration test to verify that our HasMany repository correctly filters the records, and a unit test with mocked repositories to verify that the controller calls the correct HasMany repository method, then this assertion may not be needed.

Let's keep the assertion for now.